### DISCUSSION TITLE
Some technical things when installing in Ubuntu 20.04 LTS

### DISCUSSION THREAD

**Discussion by abixadamj** [mu-editor/mu#1560] [Ideas & New Features]
I use Ubuntu 20.04 LTS and Mint 20. Today I had some problems with fresh install on Ubuntu. And I realize, trat I need to install some packages to run Mu-Editor:
```
sudo apt install python3-pyqt5
sudo apt install python3-appdirs
sudo apt install python3-pyflakes
sudo apt install python3-pycodestyle
sudo apt install python3-qtconsole
sudo apt install python3-pyqt5.qsci
sudo apt install -y python3-serial
sudo apt install -y python3-pyqt5.qtserialport
sudo apt install -y python3-circuits
sudo apt install -y python3-pip
sudo apt install -y python3-semver
sudo apt install -y python3-virtualenv
sudo apt install -y python3-black
sudo pip3 install adafruit_board_toolkit
sudo pip3 install black
```
After that mu is running, **but creates virtualenv with 176 MB of libraries downloaded insight it !!!**
Even those, which are already in system!!!

My thoughts are:

1. Maybe we should add `linux_ubuntu_dependencies.sh`  file contaning those commands to repo just for another users ? I know - this will not solve all problems with Linux, but some of them...
2. Maybe there is some way to reduce the size of venv? (I have servers with LTSP and 176MB x 350 users is quite huge amout of disk...)

**Comment by carlosperate** [comment]
If you are in Ubuntu (Intel) then you shoudn't need to install any of those packages? They should have all been brought in as PyPI dependencies.
What commands are you using to install Mu?

**Comment by carlosperate** [comment]
(Doing this bit in a different respond to be able to keep threads separated)

> After that mu is running, but creates virtualenv with 176 MB of libraries downloaded insight it !!!
Even those, which are already in system!!!

This venv is used to isolate from Mu (and in some cases the system python) the dependencies used by some of the Mu modes and the dependencies used by the user.

In cases where an installer is used (Windows and macOS) this venv separates the environment (and therefore dependencies) used to launch and run Mu from the environment used by the user code. As they can install (and uninstall) packages, if these were not separated the user could accidentally uninstall (or upgrade) a Mu package and break the editor.

In cases were the system python is used this is even more important, as the user could accidentally uninstall important system packages from the Mu settings.

**Reply by carlosperate** [reply]
Looking at a macOS venv the biggest package is numpy, which is brought in as part of the iPython console.

<img width="1280" alt="image" src="https://user-images.githubusercontent.com/4189262/117952324-fa954c80-b30c-11eb-89a4-f6719479e6ed.png">

I guess one could argue that dependency is not really required for iPython to still run fine for most people. I guess it's something we could explore, not sure how integrated it is.

**Reply by abixadamj** [reply]
@carlosperate I will write more later, now I have to go outside home.
![Zrzut ekranu z 2021-05-12 12-23-02](https://user-images.githubusercontent.com/5088643/117960057-e8230f00-b31c-11eb-8c5d-0a29144f43d8.png)
